python/pyarrow/tests/test_array.py

```diff
@@ -507,6 +507,19 @@ def test_array_from_numpy_datetimeD():
     assert result.equals(expected)
 
 
+def test_array_from_py_float32():
+    data = [[1.2, 3.4], [9.0, 42.0]]
+    arr1 = pa.array(data[0], type=pa.float32())
+    arr2 = pa.array(data, type=pa.list_(pa.float32()))
+
+    expected1 = np.array(data[0], dtype=np.float32)
+    expected2 = pd.Series([np.array(data[0], dtype=np.float32),
+                           np.array(data[1], dtype=np.float32)])
+
+    assert arr1.equals(pa.array(expected1))
```

Comment:
Do we want to check that `arr1.type` is exactly `pa.float32()` here?

Reply:
That's included in the call to `equals`, but I'll add an explicit check